docs/versioning.rst

```diff
@@ -0,0 +1,96 @@
+**********
+Versioning
+**********
+
+
+Current Status
+==============
+
+When a new major version is released it becomes the current release line, where all new features, bug fixes and security patches land.
+
+
+.. list-table:: Release Lines
+   :widths: 20 30 30 20
+   :header-rows: 1
+
+   * - Line
+     - Supported Runtimes
+     - Minimum Trace Agent
+     - Status
+   * - 0.x
+     - CPython: 2.7, 3.5+
+     - 
+     - MAINTENANCE [2022-02-25]
+   * - 1.x
+     - CPython: 2.7, 3.5+
+     - 7.22
+     - CURRENT
+   * - 2.x
+     - TBD
+     - TBD
+     - DEVELOPMENT
+
+
+The **status** of each release line determines when changes are made:
+
+MAINTENANCE
+    No longer under development.
+
+    Receives all bug fixes and security fixes.
+
+    Will not receive any new features.
+
+CURRENT
+    Under active development.
+
+    Receives all backwards compatible new features, bug fixes, and security fixes.
+
+DEVELOPMENT
+    Under active development.
+
+    Receives all incompatible new features.
+
+    Not yet released or maintained.
+
+FUTURE
+    Planned but not under active development.
+
+Interface
+=========
+
+The definition of the **internal** interface:
+    Any module, function, class or attribute is internal if it is prefixed with a single underscore
+
+    Any package that is named ``internal`` is internal
```

Comment:
```suggestion
    Any module that is named ``internal`` is internal
```

? or do we say "package" because we meant he collection of modules under the name `*.internal` ? 🤔

Reply:
We would say "package" here. The spec for internal "module" is covered by the first item (files are prefixed with single underscore). The spec for internal "package" is any package directory that is named as such. We could also say any directory with a prefix of single underscore but I do not see that in our current directory structure.